lib/Zonemaster/Backend/Translator.pm

```diff
@@ -45,17 +45,13 @@ sub translate_tag {
         $ENV{LC_ALL} || $ENV{LC_CTYPE};
     }
 
-    my $string = $self->data->{ $entry->{module} }{ $entry->{tag} };
-
-    if ( not $string ) {
-        return $entry->{string};
-    }
-
-    my $blessed_entry = bless($entry, 'Zonemaster::Engine::Logger::Entry');
-    my $octets = Zonemaster::Engine::Translator::translate_tag( $self, $blessed_entry );
+    my $entry = Zonemaster::Engine::Logger::Entry->new( %{ $hashref } );
+    my $octets = Zonemaster::Engine::Translator::translate_tag( $self, $entry );
     $self->locale( $previous_locale );
-    my $str = decode_utf8( $octets );
-    return $str;
+
+    return ( defined $octets )
+      ? decode_utf8( $octets )
+      : $entry->string;
```

Comment:
The following is Zonemaster::Engine::Translator::translate_tag:

```
sub translate_tag {
    my ( $self, $entry ) = @_;

    return $self->_translate_tag( $entry->module, $entry->tag, $entry->printable_args ) // $entry->string;
}
```

Won't it already return $entry->string if translation (_translate_tag) fails?

Reply:
It will. I'll make an update.